packages/zoe/src/contracts/newSwap/swap.js

```diff
@@ -27,201 +27,161 @@ export const makeMakeSwapInvitation = (
   getPriceGivenRequiredOutput,
   protocolSeat,
 ) => {
-  function stageProtocolSeatFee(protocolFee) {
-    const protocolStage = protocolSeat.stage({
-      RUN: AmountMath.add(
-        protocolSeat.getAmountAllocated('RUN', protocolFee.brand),
-        protocolFee,
-      ),
-    });
-    return protocolStage;
-  }
-
-  function poolStagingFromCentral(amountIn, amountOut, protocolFee) {
+  const reallocateCentralToSecondary = (
+    seat,
+    protocolFee,
+    amountIn,
+    amountOut,
+  ) => {
     const pool = getPool(amountOut.brand);
-    // gains (amountIn - protocolFee), loses amountOut
-    return pool.stageSeat({
-      Central: AmountMath.add(
-        pool.getCentralAmount(),
-        AmountMath.subtract(amountIn, protocolFee),
-      ),
-      Secondary: AmountMath.subtract(pool.getSecondaryAmount(), amountOut),
-    });
-  }
+    const poolSeat = pool.getPoolSeat();
+
+    // Transfer amountIn from the user
+    seat.decrementBy({ In: amountIn });
+    poolSeat.incrementBy({ Central: amountIn });
+
+    // Transfer protocolFee
+
+    // TODO: this should be transferred directly from the user such
+    // that there's no way to accidentally take more than expected
+    // from the pool.
```

Comment:
Sorry, I didn't mark this copy of the TODO. I think it should also be dropped.

Reply:
Ah, I left it in purposefully. In this case, it is central to secondary, so we can take the protocolFee directly from the user's give